File: tests/fixtures/sample.lua
-- tests/fixtures/sample.lua

local M = {}

---@class Config
---@field debug boolean
local defaults = {
	debug = false,
	timeout = 500,
}

function M.setup(opts)
	opts = opts or {}
	local config = vim.tbl_deep_extend("force", defaults, opts)

	if config.debug then
		print("zitchdog: debug mode enabled")
	end

	return config
end

local function fetch(url, callback)
	local ok, err = pcall(function()
		vim.notify("Fetching " .. url)
		callback(nil, { status = 200 })
	end)

	if not ok then
		error("fetch failed: " .. err)
	end
end

return M

File: tests/fixtures/sample.py
# tests/fixtures/sample.py

from dataclasses import dataclass
from typing import Optional


@dataclass
class Config:
    debug: bool = False
    timeout: int = 500


class Fetcher:
    def __init__(self, base_url: str) -> None:
        self.base_url = base_url

    async def fetch(self, path: str, retries: int = 3) -> Optional[dict]:
        for attempt in range(retries):
            try:
                response = await self._request(path)
                return response
            except ConnectionError as err:
                print(f"attempt {attempt} failed: {err}")
        return None

    def _request(self, path: str):
        raise NotImplementedError


if __name__ == "__main__":
    config = Config(debug=True)
    print(f"Running with timeout={config.timeout}")
